leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java

```diff
@@ -29,23 +36,116 @@
  */
 public class WriteRequest extends AbstractDownlinkRequest<WriteResponse> {
 
+    public enum Mode {
+        REPLACE, UPDATE
+    }
+
     private final LwM2mNode node;
     private final ContentFormat contentFormat;
+    private final Mode mode;
+
+    // ***************** write object ****************** //
+
+    public WriteRequest(final Mode mode, final ContentFormat contentFormat, final int objectId,
```

Comment:
I'm ok with @sophokles73 about immutability.
About static constructor, this will not be homogeneous with other requests and I'm not sure to see the benefits in this case,
I will integrated this in master (I have other request blocked by this one), but the discussion is still opened.